metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -293,11 +300,47 @@ private void open() throws MetaException {
     boolean useFramedTransport = conf.getBoolVar(ConfVars.METASTORE_USE_THRIFT_FRAMED_TRANSPORT);
     int clientSocketTimeout = conf.getIntVar(ConfVars.METASTORE_CLIENT_SOCKET_TIMEOUT);
 
+    SocketFactory socketFactory = SocketFactory.getDefault();
+    Socket socket = null;
+    try {
+      socket = socketFactory.createSocket();
+      socket.setKeepAlive(true);
+
+      String hiveHost = conf.getVar(ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST);
+      if (hiveHost.isEmpty()) {
+        UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
+        if (loginUser != null && loginUser.hasKerberosCredentials()) {
+          hiveHost = SecurityUtil.getHostFromPrincipal(loginUser.getUserName());
+        }
+      }
+
+      if (!hiveHost.isEmpty()) {
```

Comment:
You're right, but Ryan was making a somewhat different point.  To address that point: it should _not_ be an "else" because the value of hiveHost may change in the first "if".